src/app/seamlessupdate/client/Settings.java

```diff
@@ -51,6 +52,24 @@ public void onCreate(final Bundle savedInstanceState) {
         PreferenceManager.setDefaultValues(createDeviceProtectedStorageContext(), R.xml.settings, false);
         addPreferencesFromResource(R.xml.settings);
 
+        final Preference checkForUpdates = findPreference(KEY_CHECK_FOR_UDPATES);
+        checkForUpdates.setOnPreferenceClickListener((final Preference preference) -> {
+            if (!getPreferences(this).getBoolean(KEY_WAITING_FOR_REBOOT, false)) {
+                PeriodicJob.cancel(this);
```

Comment:
It would be better to add a boolean flag to force `schedule` to reschedule rather than cancelling it. It's theoretically possible that the program gets killed right here.